living_documentation_regime/living_documentation_generator.py

```diff
@@ -281,460 +264,29 @@ def _consolidate_issues_data(
-        @param repository_id: The repository id of a repository that stores the issues.
+        @param output_formats: A list of output formats to be generated.
         @return: None
         """
-        index_level_dir = ""
-        replacement = {
-            "date": datetime.now().strftime("%Y-%m-%d"),
-        }
-
-        # Set correct behaviour based on the level
-        if level == "org":
-            organization_name = repository_id.split("/")[0]
-            index_level_dir = organization_name
-            replacement["organization_name"] = organization_name
-        elif level == "repo":
-            repository_name = repository_id.split("/")[1]
-            index_level_dir = repository_id
-            replacement["repository_name"] = repository_name
-
-        # Replace the issue placeholders in the index template
-        sub_level_index_page = index_template.format(**replacement)
-
-        # Create a sub index page file
-        output_path = os.path.join(make_absolute_path(LIV_DOC_OUTPUT_PATH), index_level_dir)
-        with open(os.path.join(output_path, "_index.md"), "w", encoding="utf-8") as f:
-            f.write(sub_level_index_page)
-
-    @staticmethod
-    def _generate_markdown_line(consolidated_issue: ConsolidatedIssue) -> str:
-        """
-        Generates a markdown summary line for a single issue.
-
-        @param consolidated_issue: The ConsolidatedIssue object containing the issue data.
-        @return: The markdown line for the issue.
-        """
-        organization_name = consolidated_issue.organization_name
-        repository_name = consolidated_issue.repository_name
-        number = consolidated_issue.number
-        title = consolidated_issue.title
-        title = title.replace("|", " _ ")
-        issue_link_base = consolidated_issue.title.replace(" ", "-").lower()
-        issue_mdoc_link = f"features#{issue_link_base}"
-        url = consolidated_issue.html_url
-        state = consolidated_issue.state
-
-        status_list = [project_status.status for project_status in consolidated_issue.project_issue_statuses]
-        status = ", ".join(status_list) if status_list else "---"
-
-        # Change the bool values to more user-friendly characters
-        if ActionInputs.get_is_project_state_mining_enabled():
-            if consolidated_issue.linked_to_project:
-                linked_to_project = LINKED_TO_PROJECT_TRUE
-            else:
-                linked_to_project = LINKED_TO_PROJECT_FALSE
-
-            # Generate the Markdown issue line WITH extra project data
-            md_issue_line = (
-                f"| {organization_name} | {repository_name} | [#{number} - {title}]({issue_mdoc_link}) |"
-                f" {linked_to_project} | {status} |<a href='{url}' target='_blank'>GitHub link</a> |\n"
-            )
+        # If list of output formats is by user an empty list, generate all output options
+        if not output_formats:
```

Comment:
Get data from ActionInputs here.
Call static method on Factory class and get instances of generator by required output formats.
Call export method on each of them and provide mined data.

If correct validation done in ActionInputs - in initial run of logic no more check for no value needed.